registrations/models.py

```diff
@@ -207,6 +207,43 @@ def fire_language_metric(sender, instance, created, **kwargs):
         })
 
 
+@receiver(post_save, sender=Registration)
+def fire_state_metric(sender, instance, created, **kwargs):
+    """
+    Fires metrics for each state for each subscription, a sum metric for
+    the registrations over time, and a last metric for the total count.
+    """
+    from .tasks import fire_metric, is_valid_state
+    from hellomama_registration.utils import (get_identity, search_identities,
+                                              normalise_string)
+    if (created and instance.data and instance.data['operator_id']):
+        identity = get_identity(instance.data['operator_id'])
+        if (identity['details'] and identity['details']['state'] and
```

Comment:
This should probably be more something like `identity.get('details')` and `identity['details'].get('state')` to avoid value errors. (I'm assuming the identity is a `dict`)